test/e2e/streaming_test.go

```diff
@@ -50,6 +53,13 @@ func (suite *StreamingTestSuite) SetupTest() {
 	t = suite.T()
 }
 
+func (suite *StreamingTestSuite) AfterTest(suiteName, testName string) {
+	if debugMode && t.Failed() {
```

Comment:
This is repeated all over, perhaps it would make sense to move it to a func, like,` func handleFailure(t *testing.T)` ? Then all you call from the tests is `handleFailure(t)`.